packages/alchemy/src/config/utils/deserialize.ts

```diff
@@ -0,0 +1,18 @@
+// https://github.com/wevm/wagmi/blob/main/packages/core/src/utils/deserialize.ts
+type Reviver = (key: string, value: any) => any;
+
+/**
+ * JSON parses a string while correctly handling BigInt and Map types.
+ *
+ * @param value the string to deserialize
+ * @param reviver optional reviver function to transform the parsed value
+ * @returns the parsed object
+ */
+export function deserialize<type>(value: string, reviver?: Reviver): type {
```

Comment:
I think it might be worth putting a little more detail into the doc-comment here to clarify how this should be used:

* Where is that input format coming from with the `{ value, __type }` objects? I didn't see a corresponding `serialize` function (but maybe I missed it).
* That this function calls `decodeURIComponent` on its input is highly relevant to callers and might be worth noting in the doc-comment. (If this is part of the corresponding serialize function then possibly doesn't need a separate mention.)